dbms/src/Operators/Operator.cpp

```diff
@@ -36,31 +50,35 @@ OperatorStatus Operator::await()
     // `exec_status.is_cancelled` has been checked by `EventTask`.
     // If `exec_status.is_cancelled` is checked here, the overhead of `exec_status.is_cancelled` will be amplified by the high frequency of `await` calls.
 
-    // TODO collect operator profile info here.
     auto op_status = awaitImpl();
 #ifndef NDEBUG
     assertOperatorStatus(op_status, {OperatorStatus::FINISHED, OperatorStatus::NEED_INPUT, OperatorStatus::HAS_OUTPUT});
 #endif
     FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::random_pipeline_model_operator_run_failpoint);
+    // When `op_status` turns to `OperatorStatus::WAITING`, `profile_info.update()` must be called,
+    // which means that `OperatorProfileInfo::StopWatch::last_ts` has been reseted, so there is no need to call `profile_info.anchor()`.
+    if (op_status != OperatorStatus::WAITING)
+        profile_info.update();
```

Comment:
Above comments and code confuse me a little bit.

Reply:
Could the new comment be more reasonable?
```
    // During the period when op_status changes from non-waiting to waiting, and from waiting to non-waiting, await will be called continuously.
    // Therefore, directly measuring this period is equivalent to measuring the time consumed by await.
    //
    // When op_status changes to waiting, profile_info.update has already been called, meaning that profile_info::StopWatch::last_ns has been updated to that moment in time. 
    // When op_status changes to non-waiting, calling profile_info.update record the time of waiting.
    //
    //    ┌─────────────────────waiting time────────────────────────┐
    // [non-waiting, waiting, waiting, waiting, .., waiting, non-waiting]
```

Reply:
From profileInfo view like this? :
anchor, update(start waiting), update, update, update(ends waiting), anchor

Reply:
No, it is `anchor, update(start waiting), update(ends waiting), anchor`.
`update` called between `update(start waiting)` and `update(ends waiting)` is meaningless.

Reply:
ok, another question, will this waiting time be also computed in Task's getIOPendingTimeNs?

Reply:
> ok, another question, will this waiting time be also computed in Task level's statistics, like getIOPendingTimeNs or others?

yes, waiting time is also calculated in `TaskProfileInfo`.
`TaskProfileInfo.waiting_time = sum(operator[0...i].waiting_time)`